include/core/SurfaceLayer.hpp

```diff
@@ -0,0 +1,87 @@
+#ifndef __NGEN_SURFACE_LAYER__
+#define __NGEN_SURFACE_LAYER__
+
+#include "Layer.hpp"
+
+namespace ngen
+{
+    class SurfaceLayer : public Layer
+    {
+        public:
+        
+        SurfaceLayer(
+                const LayerDescription& desc, 
+                const std::vector<std::string>& p_u, 
+                const Simulation_Time& s_t, 
+                feature_type& f, 
+                geojson::GeoJSON cd, 
+                std::shared_ptr<pdm03_struct> pdmd,
+                long idx,
+                const std::vector<std::string>& n_u,
+                std::unordered_map<std::string, std::ofstream>& output_files) : 
+                    Layer(desc,p_u,s_t,f,cd,pdmd,idx), 
+                    nexus_ids(n_u), 
+                    nexus_outfiles(output_files)
+        {
+
+        }
+
+        /***
+         * @breif Run one simulation timestep for each model in this layer
+        */
+        virtual void update_models()
+        {
+            long current_time_index = output_time_index;
+            
+            Layer::update_models();
+
+            //At this point, could make an internal routing pass, extracting flows from nexuses and routing
+            //across the flowpath to the next nexus.
+            //Once everything is updated for this timestep, dump the nexus output
+            for(const auto& id : features.nexuses()) 
+            {
+                std::string current_timestamp = simulation_time.get_timestamp(current_time_index);
+                
+                #ifdef NGEN_MPI_ACTIVE
+                if (!features.is_remote_sender_nexus(id)) { //Ensures only one side of the dual sided remote nexus actually doing this...
+                #endif
+
+                //Get the correct "requesting" id for downstream_flow
+                const auto& nexus = features.nexus_at(id);
+                const auto& cat_ids = nexus->get_receiving_catchments();
+                std::string cat_id;
+                if( cat_ids.size() > 0 ) {
+                //Assumes dendridic, e.g. only a single downstream...it will consume 100%  of the available flow
```

Comment:
```suggestion
                //Assumes dendritic, e.g. only a single downstream...it will consume 100%  of the available flow
```

Reply:
This is in #494 and in https://github.com/donaldwj/ngen/pull/6 .